Neos.CliSetup/Classes/Command/SetupCommandController.php

```diff
@@ -0,0 +1,193 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\CliSetup\Command;
+
+/*
+ * This file is part of the Neos.CliSetup package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Cli\CommandController;
+use Neos\Utility\Arrays;
+use Neos\CliSetup\Exception as SetupException;
+use Neos\CliSetup\Infrastructure\Database\DatabaseConnectionService;
+use Neos\CliSetup\Infrastructure\ImageHandler\ImageHandlerService;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * @Flow\Scope("singleton")
+ */
+class SetupCommandController extends CommandController
+{
+
+    /**
+     * @var string
+     * @Flow\InjectConfiguration(package="Neos.Flow", path="core.context")
+     */
+    protected $flowContext;
+
+    /**
+     * @var DatabaseConnectionService
+     * @Flow\Inject
+     */
+    protected $databaseConnectionService;
+
+    /**
+     * @var array
+     * @Flow\InjectConfiguration(package="Neos.Flow", path="persistence.backendOptions")
+     */
+    protected $persistenceConfiguration;
+
+    /**
+     * @var ImageHandlerService
+     * @Flow\Inject
+     */
+    protected $imageHandlerService;
+
+    /**
+     * @var string
+     * @Flow\InjectConfiguration(package="Neos.Imagine", path="driver")
+     */
+    protected $imagineDriver;
+
+    /**
+     * Configure the database connection for flow persistence
+     *
+     * @param string|null $driver Driver
+     * @param string|null $host Hostname or IP
+     * @param string|null $dbname Database name
+     * @param string|null $user Username
+     * @param string|null $password Password
+     */
+    public function databaseCommand(?string $driver = null, ?string $host = null, ?string $dbname = null, ?string $user = null, ?string $password = null): void
+    {
+        $availableDrivers = $this->databaseConnectionService->getAvailableDrivers();
+        if (count($availableDrivers) == 0) {
+            $this->outputLine("No supported database driver found");
+            $this->quit(1);
+        }
+
+        if (is_null($driver)) {
+            $driver = $this->output->select(
+                sprintf('DB Driver (<info>%s</info>): ', $this->persistenceConfiguration['driver']),
+                $availableDrivers,
+                $this->persistenceConfiguration['driver']
+            );
+        }
+
+        if (is_null($host)) {
+            $host = $this->output->ask(
+                sprintf('Host (<info>%s</info>): ', $this->persistenceConfiguration['host'] ?? '127.0.0.1'),
+                $this->persistenceConfiguration['host'] ?? '127.0.0.1'
```

Comment:
just wondering is ?? the right one or is '' returned?

Reply:
This falls back to '127.0.0.1' if $this->persistenceConfiguration['host'] is null or does not exist.
See: https://www.php.net/manual/en/migration70.new-features.php#migration70.new-features.null-coalesce-op

Tested it again and it works as expected but i noticed that other empty values show empty brackets which is confusion so i added three dashes there `---`. Nut entirely sure wether `null` would be better understandable.

Reply:
nice works well ;)

three dashes are better than () i guess. But null would be fine to...